packages/site/cms-site/src/graphQLFetch/fetchInMemoryCache.ts

```diff
@@ -0,0 +1,34 @@
+const cache: Record<string, Response> = {};
+
+type Fetch = typeof fetch;
+
+export function createFetchInMemoryCache(fetch: Fetch): Fetch {
+    return async function (input: RequestInfo | URL, init?: RequestInit): Promise<Response> {
+        let cacheKey: string | undefined;
+        if (init?.cache === "no-store") {
+            //don't cache no-store requests
+        } else if (init?.method?.toUpperCase() === "GET") {
+            //cache all get requests
+            cacheKey = input.toString();
+        } else if (init?.body) {
+            const body = JSON.parse(init.body.toString());
```

Comment:
This could fail if the body isn't JSON (e.g., form data?). Should we add a try/catch?

Reply:
yes, absolutely